xff/xff.go

```diff
@@ -148,3 +147,21 @@ func New(l log.Logger, allowedSubnets []string) (grpc.StreamServerInterceptor, g
 	}
 	return streamer, unaryer
 }
+
+func XFFHandler(l log.Logger, mux *http.ServeMux, allowedSubnets []string) http.Handler {
+	var handler http.Handler
+	if len(allowedSubnets) > 0 {
+		xffmw, err := xff.New(xff.Options{
+			AllowedSubnets: allowedSubnets,
+		})
+		if err != nil {
+			l.Fatal(err, "error creating a new xff handler")
+		}
+
+		handler = xffmw.Handler(mux)
+	} else {
+		handler = mux
```

Comment:
I think it would be nice to handle `mux` being nil and then using `http.DefaultServeMux`